filters/openpolicyagent/authorizewithregopolicy/authorizewithregopolicy.go

```diff
@@ -0,0 +1,160 @@
+package authorizewithregopolicy
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/zalando/skipper/filters"
+	"gopkg.in/yaml.v2"
+
+	"github.com/zalando/skipper/filters/openpolicyagent"
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/envoy"
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/util"
+)
+
+type spec struct {
+	registry *openpolicyagent.OpenPolicyAgentRegistry
+	opts     []func(*openpolicyagent.OpenPolicyAgentInstanceConfig) error
+}
+
+func NewAuthorizeWithRegoPolicySpec(registry *openpolicyagent.OpenPolicyAgentRegistry, opts ...func(*openpolicyagent.OpenPolicyAgentInstanceConfig) error) filters.Spec {
+	return &spec{
+		registry: registry,
+		opts:     opts,
+	}
+}
+
+func (s *spec) Name() string {
+	return filters.AuthorizeWithRegoPolicyName
+}
+
+func (s *spec) CreateFilter(config []interface{}) (filters.Filter, error) {
+	var err error
+
+	sargs, err := util.GetStrings(config)
```

Comment:
Do we need this helper for two args? Maybe just typecast in-place like
```go
bundleName, ok := args[0].(string)
if !ok {
  return nil, filters.ErrInvalidFilterParameters
}
```

Reply:
I'll remove it...